ai_server/server.py

```diff
@@ -67,27 +72,38 @@ def is_llamacpp_available(model: str) -> bool:
     """Check if model is available in llama.cpp."""
     return resolve_model_path(model) is not None
 
-def chat_with_ollama(model: str, content: str) -> str:
+def chat_with_ollama(model: str, content: str, system_prompt: Optional[str] = None) -> str:
     """Handle chat using ollama."""
+    messages = []
+    if system_prompt:
+        messages.append({'role': 'system', 'content': system_prompt})
+    messages.append({'role': 'user', 'content': content})
+    
     response = ollama.chat(
         model=model,
-        messages=[{'role': 'user', 'content': content}],
+        messages=messages,
         stream=False
     )
     return response.message.content
 
-def chat_with_llamacpp(model: str, content: str, timeout: int = 300) -> str:
+def chat_with_llamacpp(model: str, content: str, system_prompt: Optional[str] = None, timeout: int = 300) -> str:
     """Handle chat using llama.cpp CLI."""
     model_path = resolve_model_path(model)
     
     if not model_path:
         raise ValueError(f"Model not found: {model}")
     
+    # Format prompt with system prompt if provided
```

Comment:
There is a separate argument that you can use instead of modifying the user prompt. See https://github.com/ggml-org/llama.cpp/tree/master/tools/main